### PR TITLE
refactor(ui): leverage react-helmet to manage headers

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

leverage react-helmet to manage headers, HTML and body attrs

image

- favicon
- apple-touch-icon
- lang attribute
- data-color-mode attribute
- custom CSS style 
- body platform class
- body color theme class

## Testing

test locally

## Checklist

- [ ] `.changset`
- [ ] unit tests
- [ ] integration tests
- [ ] docs

OR

- [x] This PR is not applicable for the checklist

### CODE REVIEW

File: packages/ui/src/containers/VerificationCode/index.test.tsx [hunk 1]
```diff
@@ -244,8 +244,6 @@ describe('<VerificationCode />', () => {
           verificationCode: '111111',
         });
       });
-
-      // TODO: @simeng test exception flow to fulfill the password
```
Comment (simeng-li) [inline packages/ui/src/containers/VerificationCode/index.test.tsx:248]: had an issue tracking this

File: packages/ui/src/hooks/use-sie.ts [hunk 2]
```diff
@@ -19,7 +19,16 @@ export const useSieMethods = () => {
     socialConnectors: experienceSettings?.socialConnectors ?? [],
     signInMode: experienceSettings?.signInMode,
     forgotPassword: experienceSettings?.forgotPassword,
-    customContent: experienceSettings?.customContent,
+  };
+};
+
+export const useCustomSie = () => {
```
Comment (gao-sun) [inline packages/ui/src/hooks/use-sie.ts:25]: how about we use one hook for all props as we can pick in the component? this should be good for now (as they are staying in the same context)

```ts
const useSignInExperience = () => {};
```

File: packages/ui/src/Providers/AppBoundary/AppMeta.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import defaultFavicon from '@/assets/favicon.png';
+
+import * as styles from './index.module.scss';
+
+/**
+ * User React Helmet to manage html and body attributes
+ * @see https://github.com/nfl/react-helmet
+ *
+ * - lang: set html lang attribute
+ * - data-color-mode: set html data-color-mode attribute
```
Comment (gao-sun) [inline packages/ui/src/Providers/AppBoundary/AppMeta.tsx:18]: cloud not find reference for this, do you have a reference link handy?

Comment (simeng-li) [reply packages/ui/src/Providers/AppBoundary/AppMeta.tsx:18]: Not really, simply provide a new way of color theme ref for those who want to set theme-based custom CSS.  Checked GitHub is using these attributes for their theme-based styles.

File: packages/ui/src/hooks/use-sie.ts (hunk 2 above)
Comment (simeng-li) [reply packages/ui/src/hooks/use-sie.ts:25]: Could, kind of forgot why I split them up.

File: packages/ui/src/Providers/AppBoundary/AppMeta.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+ * - body class: set theme body class
+ * - custom css: set custom css style tag
+ */
+
+const AppMeta = () => {
+  const { experienceSettings, theme, platform } = useContext(PageContext);
+
+  return (
+    <Helmet>
+      <html lang={i18next.language} data-color-mode={theme} />
```
Comment (gao-sun) [inline packages/ui/src/Providers/AppBoundary/AppMeta.tsx:31]:
```suggestion
      <html lang={i18next.language} data-theme={theme} />
```

File: packages/ui/src/Providers/AppBoundary/AppMeta.tsx (hunk 3 above)
Comment (gao-sun) [inline packages/ui/src/Providers/AppBoundary/AppMeta.tsx:18]:
```suggestion
 * - theme set html data-theme attribute
```

File: packages/ui/src/hooks/use-sie.ts (hunk 2 above)
Comment (simeng-li) [reply packages/ui/src/hooks/use-sie.ts]: updated

**Pull request merged**